cmd/nerdctl/run.go

```diff
@@ -638,20 +644,70 @@ func runAction(clicontext *cli.Context) error {
 		return err
 	}
 	if code != 0 {
-		return cli.NewExitError("", int(code))
+		return ExitCodeError{
+			exitCode: int(code),
+		}
 	}
 	return nil
 }
 
-func generateRootfsOpts(ctx context.Context, client *containerd.Client, clicontext *cli.Context, id string) ([]oci.SpecOpts, []containerd.NewContainerOpts, *imgutil.EnsuredImage, error) {
-	imageless := clicontext.Bool("rootfs")
+func getNetworkSlice(cmd *cobra.Command) ([]string, error) {
+	var netSlice = []string{}
+	var networkSetted = false
+	if cmd.Flags().Lookup("network").Changed {
+		network, err := cmd.Flags().GetStringSlice("network")
+		if err != nil {
+			return nil, err
+		}
+		netSlice = append(netSlice, network...)
+		networkSetted = true
+	}
+	if cmd.Flags().Lookup("net").Changed {
+		net, err := cmd.Flags().GetStringSlice("net")
+		if err != nil {
+			return nil, err
+		}
+		netSlice = append(netSlice, net...)
+		networkSetted = true
+	}
+
+	if !networkSetted {
+		network, err := cmd.Flags().GetStringSlice("network")
```

Comment:
Why can't we just call `GetStringSlice("network")` unconditionally?

Reply:
If user set network flag, use value from network; if user set net, use value from net; otherwise, use default value. So, can't call `GetStringSlice("network")` unconditionally.